overlord/snapstate/handlers.go

```diff
@@ -3500,14 +3500,57 @@ func (m *SnapManager) doSetAutoAliases(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if !snapst.AliasesPending && len(curAliases) != 0 {
+		// remove-aliases didn't run and we had existing aliases
+		// set this flag to let setup-aliases know that it should
+		// prune old aliases.
+		t.Set("prune-old-aliases", true)
+	}
+
 	t.Set("old-aliases-v2", curAliases)
+	t.Set("old-aliases-pending", snapst.AliasesPending)
 	// noop, except on first install where we need to set this here
```

Comment:
this comment is not quite true anymore I think

Reply:
updated, thanks!